### PR TITLE
build: save Docker build cache in GH Artifacts and GHA cache

### PR DESCRIPTION
**Pull request by minhlai**
# SC-1949

## Proposed changes

- add a workflow that:
  - builds the docker image and exports the docker build cache to a directory
  - uploads the docker build cache and image manifest to GHA Artifacts
  - saves the docker build cache to GHA cache with the image digest as key

---

## Reviewer notes

See this workflow summary: https://github.com/USSF-ORBIT/ussf-portal-client/actions/runs/4886817479
to see what the artifact looks like.

### Start the system
```sh
yarn services:up
yarn dev
cd ../ussf-portal-cms
yarn dev
```

Login to the portal http://localhost:3000

### Start storybook

```sh
yarn storybook
```

Login to storybook http://localhost:6006, though the command above should open it for you

---

### As the original developer, I have

- [ ] Met the acceptance criteria
- [ ] Created new stories in Storybook if applicable
- [ ] Created/modified automated unit tests in Jest
- [ ] Created/modified automated [E2E tests](https://github.com/USSF-ORBIT/ussf-portal)
- [ ] Followed guidelines for zero-downtime deploys, if applicable
- [ ] Use [ANDI](https://www.ssa.gov/accessibility/andi/help/install.html) to check for basic a11y issues

### As a reviewer, I have

Check out our [How to review a pull request](https://github.com/USSF-ORBIT/ussf-portal/blob/main/docs/how-to/review-pull-request.md) document.

---

### CODE REVIEW

File: .github/workflows/build-docker-cache-artifacts.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      - '**.md**'
+    branches: [main]
+
+jobs:
+  build-and-push-dev-aws:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3
+      - name: Get tag
+        run: echo "RELEASE_VERSION=${GITHUB_REF:10}" >> $GITHUB_ENV
```
Comment (gidjin) [inline .github/workflows/build-docker-cache-artifacts.yml:17]: Question: So this pulls in the most recent released version for testing? Or does this pull in what is latest on main for testing?

File: .github/workflows/build-docker-cache-artifacts.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        run: |
+          content=`cat /tmp/portal-client/index.json`
+          echo "index=$content" >> $GITHUB_OUTPUT
+
+      - name: Set sha digest of docker manifest
+        id: set_sha
+        env:
+          DIGEST: ${{fromJson(steps.set_var.outputs.index).manifests[0].digest}}
+        run: |
+          echo "sha256=${DIGEST##sha256:}" >> $GITHUB_OUTPUT
```
Comment (gidjin) [inline .github/workflows/build-docker-cache-artifacts.yml:61]: Thought: Is the digest all we need from the `index.json`? If so would it be simpler to consolidate this step and the previous step?

**Review by gidjin — APPROVED** [review]
Left a couple comments but this LGTM. Maybe a walkthrough at eng sync would be good?

File: .github/workflows/build-docker-cache-artifacts.yml (hunk 2 above)
Comment (minhlai) [reply .github/workflows/build-docker-cache-artifacts.yml:61]: That's a good idea! I'll try to combine the steps.

**Pull request merged**